analysis/src/python/data_collection/api/platform_api.py

```diff
@@ -0,0 +1,108 @@
+import urllib
+from dataclasses import asdict
+from typing import List, Type, Optional, TypeVar
+
+import requests
+from dacite import from_dict
+
+from analysis.src.python.data_collection.api.platform_entities import RequestParams, Object, Response
+from analysis.src.python.data_collection.utils.csv_utils import save_objects_to_csv
+from analysis.src.python.data_collection.utils.json_utils import kebab_to_snake_case
+
+T = TypeVar('T', bound=Object)
+
+
+# Base class for hyperskill and stepik clients which wraps data exchange process with platforms according to open APIs.
+class PlatformClient:
+
+    def __init__(self, host: str, token: str = None):
+        self.host = host
+        self.token = token
+
+    def _fetch(self,
+               obj_class: str,
+               params: RequestParams,
+               obj_response_type: Type[Response[T]],
+               obj_id: int = None) -> Optional[Response[T]]:
+        api_url = '{}/api/{}s'.format(self.host, obj_class, obj_id)
+        if obj_id is not None:
+            api_url = '{}/{}'.format(api_url, obj_id)
+        if params is not None:
+            dict_params = {k: v for k, v in asdict(params).items() if v is not None}
+            api_url = '{}?{}'.format(api_url, urllib.parse.urlencode(dict_params))
+        if self.token is not None:
+            raw_response = requests.get(api_url, headers={'Authorization': 'Bearer ' + self.token})
+        else:
+            raw_response = requests.get(api_url)
+
+        if raw_response.status_code == 200:
+            response_json = raw_response.json()
+        else:
+            print(f"Failed to fetch {api_url}: {raw_response}")
+            return None
+
+        preprocessed_response = kebab_to_snake_case(response_json)
```

Comment:
Sorry, but what is mean `kebab`?

Reply:
`kebab` is naming style for keys like "hello-world" (like snake "hello_world" and camel "helloWorld").

Reply:
wow, thank you!